### PR TITLE
fix fused-moe in triton2.2.0

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: lmdeploy/pytorch/kernels/fused_moe.py [hunk 1]
```diff
@@ -269,7 +269,7 @@ def _start_end_kernel(TopkIdx, SortedIdx, ExpStart, ExpEnd,
         sidx = tl.load(SortedIdx + sidx_off, mask=sidx_mask, other=0)
         tidx = tl.load(TopkIdx + sidx, mask=sidx_mask, other=num_experts)
         tidx_mask = tidx == exp_id
-        cnt += tl.sum(tidx_mask)
+        cnt += tl.sum(tidx_mask.to(tl.int32))
```
Comment (lvhan028) [inline lmdeploy/pytorch/kernels/fused_moe.py:272]: Is it compatible with triton 2.1.0?

Comment (grimoire) [reply lmdeploy/pytorch/kernels/fused_moe.py:272]: Yes, tested.

**Comment by zhulinJulia24** [comment]
fixed

**Review by zhulinJulia24 — APPROVED** [review]
LGTM

**Pull request merged**